src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -6072,6 +6063,42 @@ protected override bool IsItemContainer(string path)
             }
         }
 
+        /// <summary>
+        /// Implements the file move operation for directories without handling any error scenarios.
+        /// In particular, this attempts to rename or copy+delete the file,
+        /// but passes any exceptional behavior through to the caller.
+        /// </summary>
+        /// <param name="directory">The directory to move.</param>
+        /// <param name="destinationPath">The destination path to move the directory to.</param>
+        /// <param name="force">If true, force move the directory, overwriting anything at the destination.</param>
+        private void MoveDirectoryInfoUnchecked(DirectoryInfo directory, string destinationPath, bool force)
+        {
+#if UNIX
+            try
+            {
+                directory.MoveTo(destinationPath);
```

Comment:
I would recommend a test hook here to throw the exception with HResult == UNIX_ERRNO_EXDEV.